### PR TITLE
HUB is removing org.users and team.users

### PR DESCRIPTION
**Pull request by jctanner**
https://issues.redhat.com/browse/AAH-3371

related https://github.com/ansible/galaxy_ng/pull/2248

### CODE REVIEW

File: ansible_base/jwt_consumer/hub/auth.py [hunk 1]
```diff
@@ -27,8 +29,10 @@ def process_permissions(self):
         self.team_content_type = ContentType.objects.get_for_model(Team)
         self.org_content_type = ContentType.objects.get_for_model(Organization)
 
-        orgs = []
-        teams = []
+        admin_orgs = []
+        member_orgs = []
+        admin_teams = []
```
Comment (AlanCoding) [inline ansible_base/jwt_consumer/hub/auth.py]: Nothing is done with the top 3 relationship right now, right? Maybe a TODO comment would be good.

File: ansible_base/jwt_consumer/hub/auth.py [hunk 2]
```diff
@@ -51,25 +58,37 @@ def process_permissions(self):
...
-        team_pks = [team.pk for team in teams]
-        for team in Team.objects.exclude(pk__in=team_pks).filter(users=self.common_auth.user):
-            team.users.remove(self.common_auth.user)
-        for team in teams:
-            team.users.add(self.common_auth.user)
+        member_team_pks = [team.pk for team in member_teams]
+        # delete all memberships not defined by this jwt ...
+        RoleUserAssignment.objects.filter(
+            user=self.common_auth.user
+        ).exclude(object_id__in=member_team_pks).delete()
```
Comment (AlanCoding) [inline ansible_base/jwt_consumer/hub/auth.py]: bulk deletions won't trigger signals

**Comment by AlanCoding** [comment]
My alternative proposal was https://github.com/ansible/django-ansible-base/pull/583, I started down the path of patching up the specific logic, but it just didn't seem worth it. The only challenge is testing that.

File: ansible_base/jwt_consumer/hub/auth.py [hunk 3]
```diff
@@ -51,25 +58,38 @@ def process_permissions(self):
                         team = Resource.objects.get(ansible_id=ansible_id).content_object
                     except Resource.DoesNotExist:
                         team = self.common_auth.get_or_create_resource('team', team_data)[1]
-                    teams.append(team)
 
-                    groups.append(team.group)
+                groups.append(team.group)
+
+                if role_name == 'Team Member':
+                    member_teams.append(team)
```
Comment (john-westcott-iv) [inline ansible_base/jwt_consumer/hub/auth.py]: Shouldn't this be indented under the `for` loop?

Comment (jctanner) [reply ansible_base/jwt_consumer/hub/auth.py]: yep

File: ansible_base/jwt_consumer/hub/auth.py [hunk 4]
```diff
@@ -51,25 +58,38 @@ def process_permissions(self):
...
-        for team in teams:
-            team.users.add(self.common_auth.user)
+        member_team_pks = [team.pk for team in member_teams]
+        # delete all memberships not defined by this jwt ...
+        for assignment in RoleUserAssignment.objects.filter(
+            user=self.common_auth.user
+        ).exclude(object_id__in=member_team_pks):
+            assignment.delete()
+        # assign "local" membership for each team ...
+        roledef = RoleDefinition.objects.get(name='Galaxy Team Member')
```
Comment (AlanCoding) [inline ansible_base/jwt_consumer/hub/auth.py]: I'd like to change "Galaxy Team Member" -> "Team Member". The latter is owned by JWT auth.

File: ansible_base/jwt_consumer/hub/auth.py [hunk 5]
```diff
@@ -51,25 +58,38 @@ def process_permissions(self):
...
+        for team in member_teams:
+            RoleUserAssignment.objects.get_or_create(
+                user=self.common_auth.user,
+                role_definition=roledef,
+                object_id=team.pk
+            )
```
Comment (AlanCoding) [inline ansible_base/jwt_consumer/hub/auth.py]: Can you replace the `get_or_create` with `roledef.give_permission(user, team)`? Sorry, I said signals before but it's not signals that DAB RBAC relies on. It relies on the code that is ran outright by `give_permission`

File: ansible_base/jwt_consumer/hub/auth.py (hunk 4 above)
Comment (jctanner) [reply ansible_base/jwt_consumer/hub/auth.py]: done

**Review by AlanCoding — APPROVED** [review]
Should work for the currently supported roles.

File: ansible_base/jwt_consumer/hub/auth.py [hunk 6]
```diff
@@ -4,45 +4,42 @@
...
 
-        orgs = []
-        teams = []
-        groups = []
-        for role_name in self.common_auth.token.get('object_roles', {}).keys():
-            if role_name.startswith('Org'):
-                for object_index in self.common_auth.token['object_roles'][role_name]['objects']:
-                    org_data = self.common_auth.token['objects']['organization'][object_index]
-                    ansible_id = org_data['ansible_id']
+        # TODO - galaxy does not have an org admin roledef yet
```
Comment (AlanCoding) [inline ansible_base/jwt_consumer/hub/auth.py:30]: Thought this wasn't true, but looks like it is:

https://github.com/ansible/galaxy_ng/blob/b1c841132c4d3d27c3d0ae5ce5b744a656ef4bab/galaxy_ng/app/settings.py#L435-L437

**Review by AlanCoding — APPROVED** [review]
Thanks for finishing this up, code looks very good when viewing in the new state.

**Pull request merged**